Sources/Server/server.swift

```diff
@@ -1 +1,68 @@
+import Foundation
+import Socket
+import Requests
 
+public class Server {
+
+    var listeningSocket: Socket!
+    var clientSocket: Socket!
+    var readData = Data()
+    var parser: RequestParser
+
+    public init(parser: RequestParser) {
+        self.parser = parser
+    }
+
+    public func run() {
+        do {
+            try self.listeningSocket = Socket.create()
+            try self.listeningSocket.listen(on: 5000)
+            repeat {
+                try clientSocket = self.listeningSocket.acceptClientConnection()
+                let parsedIncomingRequest = parseRequest(socket: clientSocket)
+                let httpMethod = parsedIncomingRequest.returnMethod()
+
+                if (httpMethod == "GET" || httpMethod == "POST") {
+                    response200(socket: clientSocket)
+                } else {
+                    response404(socket: clientSocket)
+                }
+            } while true
+        } catch let error {
+            print (error.localizedDescription)
+        }
+    }
+
+    private func response200(socket: Socket) {
+        do {
+            let response = Data("HTTP/1.1 200 OK\r\nContent-Length: 0\r\n\r\n".utf8)
+            try socket.write(from: response)
+        }
+        catch let error {
+            print (error.localizedDescription)
+        }
+    }
+
+    private func response404(socket: Socket) {
+        do {
+            let response = Data("HTTP/1.1 404 NotFound\r\nContent-Length: 0\r\n\r\n".utf8)
+            try socket.write(from: response)
+        }
+        catch let error {
+            print (error.localizedDescription)
+        }
+    }
+
+    private func parseRequest(socket: Socket) -> HttpRequest {
+        do {
+            _ = try socket.read(into: &readData)
+            let incomingText = String(data: readData, encoding: .utf8)
+            let parsedRequest = try self.parser.parse(request: incomingText!)
+            readData.count = 0
+            return parsedRequest
+        } catch let error {
+            print (error.localizedDescription)
+            return HttpRequest(method: "", url: "", version: "", headers: ["" : ""])
```

Comment:
Very cool. I am not sure if you realize it here, but you are implementing the Null Object pattern. If you want to get really fancy, you can create a protocol that encompasses the methods of  HttpRequest, create a NullHttpRequest that implements the same protocol, and then use new NullHttpRequest(). Not sure how complicated that would be to implement

Reply:
Of course I didn't know there's a name for this thing. lol. I got an error and I just did whatever I can to get rid of the error :P